src/pages/Swap/index.tsx

```diff
@@ -303,7 +302,7 @@ export function Swap({
   const fiatValueOutput = useUSDPrice(parsedAmounts[Field.OUTPUT])
 
   const [routeNotFound, routeIsLoading, routeIsSyncing] = useMemo(
```

Comment:
Do we need this to be `useMemo`?

Reply:
Can we try removing `routeIsSyncing` entirely? There are some conditionals in this component that are super confusing and I think not necessary.

For example here, I think we can easily just replace with `tradeState === TradeState.LOADING`
<img width="429" alt="Screenshot 2023-05-11 at 18 36 34" src="https://github.com/Uniswap/interface/assets/2464966/d27f8b3a-cd77-4dd4-a480-cdbf02485115">
<img width="386" alt="Screenshot 2023-05-11 at 18 36 36" src="https://github.com/Uniswap/interface/assets/2464966/3ba70e07-4e08-45dd-8eaa-5d9a775a164f">
<img width="755" alt="Screenshot 2023-05-11 at 18 37 44" src="https://github.com/Uniswap/interface/assets/2464966/ab2bd5f2-be30-49f3-8b39-093a2aedfee2">

here - we have `trade` and that prop should be enough. We don't even need to pass `loading` and `syncing` because you can easily read that off `trade.state`
<img width="386" alt="Screenshot 2023-05-11 at 18 38 03" src="https://github.com/Uniswap/interface/assets/2464966/9531b862-09b0-434c-8ce8-3610c1398197">

Reply:
ill make a ticket to do this in a follow-up PR bcuz this one is getting bloated already 😓